internal/resources/k8s_deployment_resource.go

```diff
@@ -106,6 +107,20 @@ func (r *KubernetesDeploymentResource) isProgressingUpgrade() bool {
 		return true
 	}
 
+	// An update is complete when new pods are ready and old pods deleted.
+	desired := ptr.Deref(r.resource.Spec.Replicas, 2)
```

Comment:
```
// Deref dereferences ptr and returns the value it points to if no nil, or else
// returns def.
func Deref[T any](ptr *T, def T) T {
```

Why did you pick `2` here?

Reply:
because it's the [default](https://github.com/clastix/kamaji/blob/master/api/v1alpha1/tenantcontrolplane_types.go#L131). in reality this should never use the default because the field is always set, so i could just dereference. but if this code is ever unit tested it would be something to look out for